src/tests/src/study/system-model/test_component.cpp

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/unit_test.hpp>
+
+#include "antares/study/system-model/component.h"
+
+#include "utils.h"
+
+using namespace Antares::Study::SystemModel;
+
+struct ComponentTestFixture
+{
+    static Model createModel(bool with_parameters);
+};
+
+Model ComponentTestFixture::createModel(bool with_parameters)
+{
+    ModelBuilder model_builder;
+    model_builder.withId("model");
+    if (with_parameters)
+    {
+        model_builder.withParameters({Parameter("param1",
+                                                ValueType::FLOAT,
+                                                Parameter::TimeDependent::NO,
+                                                Parameter::ScenarioDependent::NO),
+                                      Parameter("param2",
+                                                ValueType::FLOAT,
+                                                Parameter::TimeDependent::NO,
+                                                Parameter::ScenarioDependent::NO)});
+    }
+    auto model = model_builder.build();
+    return model;
+}
+
+BOOST_AUTO_TEST_SUITE(_Component_)
+
+BOOST_FIXTURE_TEST_CASE(nominal_build_with_parameters, ComponentTestFixture)
+{
+    Model model = createModel(true);
+    ComponentBuilder component_builder;
+    auto component = component_builder.withId("component")
+                       .withModel(&model)
+                       .withParameterValues({{"param1", 5}, {"param2", 3}})
+                       .withScenarioGroupId("scenario_group")
+                       .build();
+    BOOST_CHECK_EQUAL(component.Id(), "component");
+    BOOST_CHECK_EQUAL(component.getModel(), &model);
```

Comment:
About :
```C++
BOOST_CHECK_EQUAL(component.getModel(), &model);
```
Rather than with its address, should we identify a **Model** with its **id** ?

Reply:
if the component identifies its model by its id, then the component should have access to the model library. then it becomes much more difficult to use

Reply:
Sorry I'm not fluent with this matter, but class **Model** has **Id()** method. Couldn't we use it here ?

Reply:
we can but it wouldn't be enough, I actually want to test that the getModel() points to the right object, not only to a model with the same ID (two different models can have the same ID if they are stored in two separate libraries)